src/nft/components/mint/NFTAssetCost.tsx

```diff
@@ -18,19 +18,19 @@ export function NFTAssetCost({ className }: NFTAssetCostReact) {
     return null;
   }
 
-  if (price?.amount === 0) {
+  if (Number(price?.amount) === 0) {
```

Comment:
`amount` is in wei right? you'll get an overflow casting this to Number

Reply:
should use native BigInt here

Reply:
no, its in ETH, converted on the BE